### PR TITLE
STYLE: dont use pd api types in tests

### LINKED ISSUE #39203: STYLE: dont' use pd.api.types anywhere in tests
example here: https://github.com/pandas-dev/pandas/pull/37367/files#diff-69c698518ca64a05f8b137ff9bb5445ce9fd21139b396e9b0d4ce7338b2a3984R132

we should just ban this with a precommit style rule (and import things directly). the doc-strings also should be updated to do the import once.

```
grep -r pd.api.types ~/pandas-dev/pandas
/home/jreback/pandas-dev/pandas/_libs/lib.pyx:    >>> pd.api.types.is_scalar(dt)
/home/jreback/pandas-dev/pandas/_libs/lib.pyx:    >>> pd.api.types.is_scalar([2, 3])
/home/jreback/pandas-dev/pandas/_libs/lib.pyx:    >>> pd.api.types.is_scalar({0: 1, 2: 3})
/home/jreback/pandas-dev/pandas/_libs/lib.pyx:    >>> pd.api.types.is_scalar((0, 2))
/home/jreback/pandas-dev/pandas/_libs/lib.pyx:    >>> pd.api.types.is_scalar(Fraction(3, 5))
/home/jreback/pandas-dev/pandas/tests/frame/test_ufunc.py:    if pd.api.types.is_extension_array_dtype(dtype) or isinstance(dtype, dict):
/home/jreback/pandas-dev/pandas/tests/frame/test_ufunc.py:        pd.api.types.is_extension_array_dtype(dtype_a)
/home/jreback/pandas-dev/pandas/tests/frame/test_ufunc.py:        or pd.api.types.is_extension_array_dtype(dtype_b)
/home/jreback/pandas-dev/pandas/tests/frame/test_ufunc.py:    if pd.api.types.is_extension_array_dtype(dtype) or isinstance(dtype, dict):
/home/jreback/pandas-dev/pandas/tests/extension/base/dtype.py:        return not pd.api.types.is_string_dtype(dtype)
/home/jreback/pandas-dev/pandas/tests/extension/base/dtype.py:        return not pd.api.types.is_object_dtype(dtype)
/home/jreback/pandas-dev/pandas/tests/extension/arrow/arrays.py:        if pd.api.types.is_scalar(item):
/home/jreback/pandas-dev/pandas/tests/extension/arrow/test_bool.py:    assert pd.api.types.is_bool_dtype(data)
/home/jreback/pandas-dev/pandas/tests/extension/json/array.py:            if pd.api.types.is_bool_dtype(item.dtype):
/home/jreback/pandas-dev/pandas/tests/extension/decimal/array.py:        if pd.api.types.is_list_like(value):
/home/jreback/pandas-dev/pandas/tests/extension/decimal/array.py:            if pd.api.types.is_scalar(key):
/home/jreback/pandas-dev/pandas/tests/extension/list/array.py:        elif pd.api.types.is_string_dtype(dtype) and not pd.api.types.is_object_dtype(
/home/jreback/pandas-dev/pandas/tests/extension/test_integer.py:                and pd.api.types.is_integer_dtype(other.dtype)
/home/jreback/pandas-dev/pandas/tests/extension/test_floating.py:                and pd.api.types.is_float_dtype(other.dtype)
/home/jreback/pandas-dev/pandas/core/dtypes/inference.py:    >>> pd.api.types.is_number(1)
/home/jreback/pandas-dev/pandas/core/dtypes/inference.py:    >>> pd.api.types.is_number(7.15)
/home/jreback/pandas-dev/pandas/core/dtypes/inference.py:    >>> pd.api.types.is_number(False)
/home/jreback/pandas-dev/pandas/core/dtypes/inference.py:    >>> pd.api.types.is_number("foo")
/home/jreback/pandas-dev/pandas/core/dtypes/inference.py:    >>> pd.api.types.is_number("5")
/home/jreback/pandas-dev/pandas/core/generic.py:        >>> cat_dtype = pd.api.types.CategoricalDtype(
```

### PR DESCRIPTION
**Pull request by nofarm3**
- [x] closes #39203 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [] whatsnew entry

I didn't add a whatsnew entry since it's only style fixes.
I'll open a new issue regarding @jreback suggestion to ban this with a precommit style rule.
I prefer it to be done in a different PR.

### CODE REVIEW

File: pandas/tests/extension/test_integer.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
 
 import pandas as pd
 import pandas._testing as tm
+from pandas.api.types import is_integer_dtype
```
Comment (jreback) [inline pandas/tests/extension/test_integer.py:23]: can you change the import on L19 as well (that is an incorrect location as well)

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. one comment. ping on green. you can add a precommit rule in this PR or as a followup.

File: pandas/tests/extension/test_integer.py (hunk 1 above)
Comment (nofarm3) [reply pandas/tests/extension/test_integer.py]: @jreback done

**Comment by MarcoGorelli** [comment]
Thanks @nofarm3 

> I'll open a new issue regarding @jreback suggestion to ban this with a precommit style rule.
I prefer it to be done in a different PR.

Any chance this could be done in the same PR, so that it can be seen to be fixed? This is analogous to adding tests in the same PR in which one fixes a bug

**Comment by nofarm3** [comment]
@MarcoGorelli 
 Yes it make sense. no problem I'll add this to this PR.

**Comment by MarcoGorelli** [comment]
sure - do ask if it's not something you're familiar with, I think it should be enough to just add a line to the `non-standard-imports-in-tests` hook

**Comment by nofarm3** [comment]
@MarcoGorelli 
I added a hook that validates there is no usage in ```pd.api.types``` under ```pandas/tests/```. 
Do you think it's good enough for now?

File: pandas/tests/extension/decimal/array.py [hunk 2]
```diff
@@ -142,8 +143,8 @@ def astype(self, dtype, copy=True):
         return super().astype(dtype, copy=copy)
 
     def __setitem__(self, key, value):
-        if pd.api.types.is_list_like(value):
-            if pd.api.types.is_scalar(key):
+        if is_list_like(value):
```
Comment (MarcoGorelli) [inline pandas/tests/extension/decimal/array.py:146]: I think the `is_list_like` import should also change to be imported from `pandas.api.types`

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Nice, thanks @nofarm3 ! Just one small comment, then I think it's good to go

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -169,6 +169,12 @@ repos:
             pandas/tests/io/excel/test_writers\.py|
             pandas/tests/io/pytables/common\.py|
             pandas/tests/io/pytables/test_store\.py$
+    -   id: no-pd-api-types
+        name: Check code for instances of pd.api.types
+        entry: pd\.api\.types
```
Comment (jreback) [inline .pre-commit-config.yaml:174]: can you make this `(pd|pandas)` (not sure if this is a full regex or not

Comment (MarcoGorelli) [reply .pre-commit-config.yaml:174]: :+1:

not sure what you mean by "full regex", but this is just `re.search`:

https://github.com/pre-commit/pre-commit/blob/ebd83171dfa8030a9767018b231b6fb1831b082b/pre_commit/languages/pygrep.py#L25-L26

Comment (nofarm3) [reply .pre-commit-config.yaml:174]: Yes, I guess I need to add it and also change the ```types``` to ```types.``` , so it won't catch the imports but only the usages in the code itself...

Example:

Good:
```from pandas.pai.types import is_number```

Not Good:
```pandas.api.types.is_number(1)```

Comment (jreback) [reply .pre-commit-config.yaml:174]: right

**Comment by nofarm3** [comment]
I'm finished, lmk if you want to see anything more.

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -166,9 +166,15 @@ repos:
         files: ^pandas/tests/
         exclude: |
             (?x)^
-            pandas/tests/io/excel/test_writers\.py
-            |pandas/tests/io/pytables/common\.py
-            |pandas/tests/io/pytables/test_store\.py$
+            pandas/tests/io/excel/test_writers\.py|
+            pandas/tests/io/pytables/common\.py|
+            pandas/tests/io/pytables/test_store\.py$
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:171]: can you revert this change please?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @nofarm3 - almost there, it just looks like an unrelated change has crept in (perhaps by accident when resolving merge conflicts?)

File: .pre-commit-config.yaml (hunk 4 above)
Comment (nofarm3) [reply .pre-commit-config.yaml]: Sorry, my mistake. done.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @nofarm3
